plugins/web-worker-offloading/hooks.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * Hook callback for Web Worker Offloading.
+ *
+ * @since n.e.x.t
+ * @package web-worker-offloading
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Gets configuration for Web Worker Offloading.
+ *
+ * @since n.e.x.t
+ * @link https://partytown.builder.io/configuration
+ *
+ * @return array{ debug?: bool, forward?: non-empty-string[], lib: non-empty-string, loadScriptsOnMainThread?: non-empty-string[], nonce?: non-empty-string } Configuration for Partytown.
+ */
+function wwo_get_configuration(): array {
+	$config = array(
+		'lib'     => wp_parse_url( plugin_dir_url( __FILE__ ), PHP_URL_PATH ) . 'build/',
+		'forward' => array(),
+	);
+
+	/**
+	 * Add configuration for Web Worker Offloading.
+	 *
+	 * @since n.e.x.t
+	 * @link https://partytown.builder.io/configuration
+	 *
+	 * @param array{ debug?: bool, forward?: non-empty-string[], lib: non-empty-string, loadScriptsOnMainThread?: non-empty-string[], nonce?: non-empty-string } $config Configuration for Partytown.
+	 */
+	return apply_filters( 'wwo_configuration', $config );
+}
+
+/**
+ * Initialize Web Worker Offloading.
+ *
+ * @since n.e.x.t
+ */
+function wwo_init(): void {
+	$partytown_js = file_get_contents( __DIR__ . '/build/partytown.js' ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- It's a local filesystem path not a remote request.
```

Comment:
could this use WP_Filesystem? (and remove the phpcs warning)

Reply:
That may be better, but core uses `file_get_contents()` for code like this:

https://github.com/WordPress/wordpress-develop/blob/9ed3553da6a61a38a1d4c5585e5ac3c9d13d351a/src/wp-includes/embed.php#L521
https://github.com/WordPress/wordpress-develop/blob/9ed3553da6a61a38a1d4c5585e5ac3c9d13d351a/src/wp-includes/formatting.php#L5962